sekoia_automation/scripts/sync_library.py

```diff
@@ -0,0 +1,311 @@
+            headers=self.headers,
+        )
+        module_uuid: str = module_info["uuid"]
+
+        if response.status_code not in (200, 404):
+            print(f"[red]Error {response.status_code}[/red]")
+
+        elif response.status_code == 404:
+            requests.post(
+                urljoin(self.symphony_url, "modules"),
+                json=module_info,
+                headers=self.headers,
+            )
+            print("Created")
+
+        else:
+            content: dict = response.json()
+            mod_list: list = list(module_info.keys())
+
+            for k in content.keys():
+                if k not in mod_list:
+                    module_info[k] = None
+
+                if content[k] == module_info[k]:
+                    del module_info[k]
+
+            if not module_info:
+                print("Already-Up-To-Date")
+            else:
+                requests.patch(
+                    urljoin(self.symphony_url, "modules", module_uuid),
+                    json=module_info,
+                    headers=self.headers,
+                )
+                print("Updated")
+
+    def load_actions(self, module_path: Path) -> list:
+        actions = []
+
+        for filename in module_path.iterdir():
+            if filename.name.endswith(".json") and filename.name.startswith("action_"):
+                action_path = module_path / filename
+                with action_path.open() as fd:
+                    actions.append(json.load(fd))
+
+        return actions
+
+    def load_triggers(self, module_path: Path) -> list:
+        triggers = []
+
+        for filename in module_path.iterdir():
+            if filename.name.endswith(".json") and filename.name.startswith("trigger_"):
+                trigger_path = module_path / filename
+                with trigger_path.open() as fd:
+                    triggers.append(json.load(fd))
+
+        return triggers
+
+    def set_docker(self, manifests: list, module: dict) -> list:
+        if "docker" in module and "version" in module:
+            for manifest in manifests:
+                if "docker" not in manifest or manifest["docker"] == module["docker"]:
+                    manifest["docker"] = f"{module['docker']}:{module['version']}"
+
+        return manifests
+
+    def get_module_logo(self, module_path: Path) -> str | None:
+        svglogo_path = module_path / "logo.svg"
+        pnglogo_path = module_path / "logo.png"
+
+        path_to_use: Path | None = None
+
+        if svglogo_path.is_file():
+            prefix = "data:image/svg+xml;base64,"
+            path_to_use = svglogo_path
+        elif pnglogo_path.is_file():
+            prefix = "data:image/png;base64,"
+            path_to_use = pnglogo_path
+
+        if path_to_use:
+            with path_to_use.open("rb") as f:
+                return f"{prefix}{b64encode(f.read()).decode('utf-8')}"
+
+        return None
+
+    def check_image_on_registry(
```

Comment:
I'm not sure about this part (how to make it work with multiple registries).
For now we can try with the current implementation :)

Reply:
👍